src/main/java/mcp/mobius/betterbarrels/common/items/dolly/ItemBarrelMover.java

```diff
@@ -177,12 +181,43 @@ public boolean onItemUseFirst(ItemStack stack, EntityPlayer player, World world,
         }
 
         if (stack.hasTagCompound() && stack.getTagCompound().hasKey("Container")) {
+            preventFold = true;
```

Comment:
Save all necessary data inside `ItemStack` and not as field of `Item`.
```suggestion
            stack.getTagCompound().setBoolean("PreventFolding", true);
```